### DISCUSSION TITLE
I can't seem to reference App State in Nativate to.

### DISCUSSION THREAD

**Discussion by jblejder** [Budibase/budibase#3841] [Help] [answered]
So I am trying to make simple secret santa app and I got stuck at very beginning.
I've read this but doesn't seem to work.
https://docs.budibase.com/building-apps/design/app-state

In this case I want to have a "host" which start session. It generates code that others reference on mobile and start a game.
So I root page with just a button. I want to generate CODE and then navigate to `/event/CODE` which will be later a place to randomly assign gifts after everyone sign up.

<img width="978" alt="Screen Shot 2021-12-21 at 8 27 17 am" src="https://user-images.githubusercontent.com/17861372/146834980-86b18f1c-f969-42c8-a222-b4f699c17e0b.png">
<img width="1013" alt="Screen Shot 2021-12-21 at 8 27 26 am" src="https://user-images.githubusercontent.com/17861372/146834990-0795ebfd-1d3b-4c57-9d38-42c8174a8013.png">

So for simple case I just create a code in App State and want to reference it in Navigate To but it doesn't work.
What am I doing wrong or how should I reference it?

**Comment by mjashanks** [comment, marked as answer]
I see what you mean. It looks like there is an issue there, whereby the State is not actually set before `Navigate To`.

Instead, you could:
1. Change your `/event/:code` URL to `/event`
2. `Update State` as above, and `Navigate to: /event`
3. Then, in the `/event` screen, use `{{ State.code }}` instead of `{{ URL.code }}`

If you must have the `/event/:code` URL for other reasons, then you could
1. `Navigate to: /event/use_state` (i.e.  `/event/any static value`)
2. In your `/event/:code` screen, when you need the code, use Javascript to do..

```javascript
if ($("URL.code") === "use_state") return $("State.Code");
else return $("URL.code") ;
```

**Reply by jblejder** [reply]
Thanks for answer. 
Yeah ok so you say it will get set after all Button actions are finished. That will work and to be able to have multiple "sessions" or different games I will probably need to clear it there and I can handle url separately. If I start setting it in DB too.

Do you think one day it will be implemented? To be able to Set State in Button and do something else with it?

In real world situation I would want button to have 1. Set App State 2. store it in DB 3. reference it in Navigate To.

**Reply by mjashanks** [reply]
I feel like it's a bug actually - I'll get it logged.
